### PR TITLE
feat: add knock off mode

### PR DESCRIPTION
**Pull request by thoa01**

### CODE REVIEW

**Review by DanHuy77 — APPROVED** [review]

File: the-telephone-app-fe/src/components/Draw/Draw.js [hunk 1]
```diff
@@ -17,13 +17,18 @@ import { IP } from "../../config/config";
 const Draw = ({ width = "815rem", height = "350rem" }) => {
   const location = useLocation();
   const turn = location.state?.turn;
-  const totalTurn = location.state?.data.length;
+  const mode = location.state?.mode;
+  let totalTurn;
   const id_room = location.state?.id_room;
```
Comment (ces-manhnguyen) [inline the-telephone-app-fe/src/components/Draw/Draw.js]: use camelCase

File: the-telephone-app-fe/src/components/Draw/Draw.js [hunk 2]
```diff
@@ -17,13 +17,18 @@ import { IP } from "../../config/config";
...
   const currentName = location.state?.name;
   const dataReceive = location.state?.dataReceive;
   const [timer, setTimer] = useState(60);
   const buttonDoneRef = useRef(null);
-
+  if (mode !== "KNOCK_OFF") {
+    totalTurn = location.state?.totalTurn;
+  } else {
+    totalTurn = location.state?.data.length;
+  }
```
Comment (ces-manhnguyen) [inline the-telephone-app-fe/src/components/Draw/Draw.js:32]: Add some spaces before and after this code block.

File: the-telephone-app-fe/src/components/Draw/Draw.js [hunk 3]
```diff
@@ -72,10 +77,15 @@ const Draw = ({ width = "815rem", height = "350rem" }) => {
...
-    );
+    if (mode === "KNOCK_OFF" && turn === 1) {
+      const response = await axios.post(
+        IP + `user/done/${id_room}/${currentName}/${image}/${turn}`
+      );
+    } else {
+      const response = await axios.post(
+        IP + `user/done/${id_room}/${dataReceive.receiver}/${image}/${turn}`
+      );
+    }
```
Comment (ces-manhnguyen) [inline the-telephone-app-fe/src/components/Draw/Draw.js]: Add some spaces before this code block.

File: the-telephone-app-fe/src/components/JoinRoom/JoinRoom.js [hunk 4]
```diff
@@ -60,13 +59,31 @@ const JoinRoom = () => {
   };
 
   const handlePlay = async () => {
-    const response = await axios.post(IP + `user/start/${id_room}`);
+    var mode1 = document.querySelector('input[value="IN_PROGRESS"]');
+    var mode2 = document.querySelector('input[value="KNOCK_OFF"]');
+    var selectedMode = null;
```
Comment (ces-manhnguyen) [inline the-telephone-app-fe/src/components/JoinRoom/JoinRoom.js]: use a state to store the current mode and update it using a function triggered by onChange, then get the mode state and use it here.

File: the-telephone-app-fe/src/components/ShowResult/ShowResult.js [hunk 5]
```diff
@@ -152,13 +153,24 @@ const ShowResult = () => {
                               </div>
                               {timer >= (index + 2) * 3 ? (
                                 <>
+                                {mode === 'KNOCK_OFF' ? (
+                                  <div className="content-img">
+                                  <ConvertUrl data={result.data} />
+                                  <div ref={scrollPoint}></div>
+                                </div>
+
+
```
Comment (ces-manhnguyen) [inline the-telephone-app-fe/src/components/ShowResult/ShowResult.js]: Fix indent and remove empty lines.

File: the-telephone-app-fe/src/components/ShowResult/ShowResult.js [hunk 6]
```diff
@@ -152,13 +153,24 @@ const ShowResult = () => {
...
-                                  </div>
+                                    {result.data
+                                      .toString()
+                                      .replace(new RegExp("_", "g"), " ")}
                                   <div ref={scrollPoint}></div>
+                                  </div>
+
+
+                                )}
+
```
Comment (ces-manhnguyen) [inline the-telephone-app-fe/src/components/ShowResult/ShowResult.js]: Remove empty lines.

File: the-telephone-app-fe/src/components/WriteSentence/WriteSentence.js [hunk 7]
```diff
@@ -14,12 +14,13 @@ import { SENTENCES } from "../../config/config";
...
   const location = useLocation();
   const id_room = location.state?.id_room;
+  const mode = location.state?.mode;
   const currentName = location.state?.name;
   const turn = location.state?.turn;
   const totalTurn = location.state?.data.length;
   const [timer, setTimer] = useState(60);
   const buttonDoneRef = useRef(null);
-
+  const dataReceive = location.state?.dataReceive;
```
Comment (ces-manhnguyen) [inline the-telephone-app-fe/src/components/WriteSentence/WriteSentence.js]: Is this unused in this file? If yes then remove it.

**Review by ces-manhnguyen — APPROVED** [review]

**Pull request merged**